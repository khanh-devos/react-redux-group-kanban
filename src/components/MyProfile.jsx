export default function MyProfile() {
  return (<div>My profile</div>);
}
