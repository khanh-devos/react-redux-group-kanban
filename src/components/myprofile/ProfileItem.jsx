import { PropTypes } from 'prop-types';

export default function ProfileItem({ name }) {
  return (<li className="reserved-rockets">{name}</li>);
}

ProfileItem.propTypes = {
  name: PropTypes.string.isRequired,
};
