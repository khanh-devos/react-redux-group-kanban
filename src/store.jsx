import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './redux/missions/missionsSlice';
import rocketReducer from './redux/rocketSlice/rocketSlice';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
    mission: missionsReducer,
  },
});

export default store;
