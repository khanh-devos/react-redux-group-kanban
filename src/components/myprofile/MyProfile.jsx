import { useSelector } from 'react-redux';
import ProfileItem from './ProfileItem';
import './profile.css';

export default function MyProfile() {
  const { rockets } = useSelector((store) => store.rocket);
  const { missions } = useSelector((store) => store.mission);

  return (
    <div className="my-profile">
      <div className="profile-mission">
        <h3 className="my-profile-booked-h3">
          My Missions
        </h3>
        <ul className="profile-ul">
          {
          missions.filter((item) => item.reserved)
            .map(({ missionId, missionName }) => <ProfileItem key={missionId} name={missionName} />)
          }
        </ul>
      </div>
      <div className="profile-rocket">
        <h3 className="my-profile-booked-h3">
          My Rockets
        </h3>
        <ul className="profile-ul">
          {
          rockets.filter((item) => item.reserved)
            .map(({ id, name }) => <ProfileItem key={id} name={name} />)
          }
        </ul>
      </div>
    </div>
  );
}
