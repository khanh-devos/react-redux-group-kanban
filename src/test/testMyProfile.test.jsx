import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import MyProfile from '../components/myprofile/MyProfile';
import ProfileItem from '../components/myprofile/ProfileItem';
import store from '../store';

describe('test Profile', () => {
  it('should match MyProfile snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match ProfileItem snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <ProfileItem name="test" />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
