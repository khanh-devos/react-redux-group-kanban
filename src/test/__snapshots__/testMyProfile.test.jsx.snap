// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Profile should match MyProfile snapshot 1`] = `
<div>
  <div
    class="my-profile"
  >
    <div
      class="profile-mission"
    >
      <h3
        class="my-profile-booked-h3"
      >
        My Missions
      </h3>
      <ul
        class="profile-ul"
      />
    </div>
    <div
      class="profile-rocket"
    >
      <h3
        class="my-profile-booked-h3"
      >
        My Rockets
      </h3>
      <ul
        class="profile-ul"
      />
    </div>
  </div>
</div>
`;

exports[`test Profile should match ProfileItem snapshot 1`] = `
<div>
  <li
    class="reserved-rockets"
  >
    test
  </li>
</div>
`;
